### ISSUE TITLE
Make BooleanParser understand "on" as true

### ISSUE THREAD

**Issue by apeiniger** [mpscholten/request-parser#46]
"on" is default value for checkboxes in html in many browsers, therefore it would be great if "on" would be treated as true as well in the BooleanParser.

**Comment by mpscholten** [comment]
Could be implemented like the [`YesNoBooleanParser`](https://github.com/mpscholten/request-parser/blob/master/src/YesNoBooleanParser.php)

**Comment by yuvalherziger** [comment]
Good point, that is indeed a missing feature, we can implement a type parser for that

**Comment by yuvalherziger** [comment]
@apeiniger @mpscholten I'm not sure because it's been a while, but on for a default checkbox button always returns `"on"` if the checkbox exists, see [example here](http://stackoverflow.com/questions/16611704/checkbox-value-always-return-on). The point is, adding an `OnOffBooleanParser` is technically possible, but is it of any added value?

PR #47 already supports it, but the question at hand is whether in terms of HTML (of which the request parser is agnostic) a form would ever submit an `off` value by default for any browser

**Comment by apeiniger** [comment]
Yeah, that was my idea as well, that's why I thought it could maybe be added to the normal BooleanParser instead, but up for discussion.
